form/src/main/java/org/dhis2/form/ui/binding/OptionSetSelectionBindings.kt

```diff
@@ -0,0 +1,118 @@
+package org.dhis2.form.ui.binding
+
+import android.content.res.ColorStateList
+import android.os.Build
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.CompoundButton
+import android.widget.ImageView
+import android.widget.LinearLayout
+import android.widget.RadioGroup
+import androidx.databinding.BindingAdapter
+import org.dhis2.form.databinding.OptionSetSelectCheckItemBinding
+import org.dhis2.form.databinding.OptionSetSelectItemBinding
+import org.dhis2.form.model.FieldUiModel
+import org.dhis2.form.model.UiRenderType
+import org.dhis2.form.ui.style.FormUiColorType
+import org.dhis2.form.ui.style.FormUiModelStyle
+
+@BindingAdapter("delete_visibility")
+fun ImageView.setOptionSetDeleteVisibility(item: FieldUiModel) {
+    visibility = when {
+        item.value != null && item.editable -> when (item.renderingType) {
+            UiRenderType.HORIZONTAL_RADIOBUTTONS,
+            UiRenderType.VERTICAL_RADIOBUTTONS -> View.VISIBLE
+            else -> View.GONE
+        }
+        else -> View.GONE
+    }
+}
+
+@BindingAdapter("renderingType")
+fun LinearLayout.setRenderingType(renderingType: UiRenderType?) {
+    when (renderingType) {
+        UiRenderType.VERTICAL_CHECKBOXES -> {
+            orientation = LinearLayout.VERTICAL
+            visibility = View.VISIBLE
+        }
+        UiRenderType.HORIZONTAL_CHECKBOXES -> {
+            orientation = LinearLayout.HORIZONTAL
+            visibility = View.VISIBLE
+        }
+        else -> visibility = View.GONE
+    }
+}
+
+@BindingAdapter("renderingType")
+fun RadioGroup.setRenderingType(renderingType: UiRenderType?) {
```

Comment:
As Radiougroup extends from LinearLayout, could these two methods be merged?

Reply:
AL moved to FieldBindigns to be reuse for other valueType?

Reply:
In this case both methods are needed because one shows/hides the checkboxes and the other the radio buttons, if merged, the behaviour is not handled correctly and I think it is a very specific behaviour for optionset only.

Reply:
ok